packages/design-system/src/components/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -0,0 +1,193 @@
+						<StackHorizontal gap="S" align="center" wrap="nowrap">
+							{'href' in origin ? (
+								<Link href={origin.href} target={origin.target} withEllipsis>
+									{origin.label}
+								</Link>
+							) : (
+								<Link as={origin.as} target={origin.target} withEllipsis>
+									{origin.label}
+								</Link>
+							)}
+
+							<span aria-hidden className={styles.divider}>
+								<Divider orientation="vertical" />
+							</span>
+						</StackHorizontal>
+					</li>
+
+					<li className={classnames(styles.entry, styles.collapsed)}>
+						<StackHorizontal gap="S" align="center" wrap="nowrap">
+							<Dropdown
+								aria-label={t('COLLAPSED_LINKS_MENU', 'Collapsed links')}
+								items={collapsed.map(collapsedLinks => {
+									if ('href' in collapsedLinks) {
+										return {
+											href: collapsedLinks.href,
+											label: collapsedLinks.label,
+											target: collapsedLinks.target,
+											type: 'link',
+										};
+									}
+									return {
+										label: collapsedLinks.label,
+										target: collapsedLinks.target,
+										type: 'link',
+										as: collapsedLinks.as,
+									};
+								})}
+							>
+								<ButtonTertiary isDropdown size="S" onClick={() => {}}>
+									<VisuallyHidden>
+										{t('COLLAPSED_LINKS_BUTTON', 'Display collapsed links')}
+									</VisuallyHidden>
+									<span aria-hidden>…</span>
+								</ButtonTertiary>
+							</Dropdown>
+							<span aria-hidden className={styles.divider}>
+								<Divider orientation="vertical" />
+							</span>
+						</StackHorizontal>
+					</li>
+
+					{suffix.map((entry, index) => {
+						const isLastEntry = index === suffix.length - 1;
+						if ('href' in entry) {
+							const { label, ...entryProps } = entry;
+							return (
+								<li className={styles.entry} key={`${label}-${index}`}>
+									<StackHorizontal gap="S" align="center" wrap="nowrap">
+										<Link
+											{...entryProps}
+											aria-current={isLastEntry ? 'page' : undefined}
+											withEllipsis
+										>
+											{label}
+										</Link>
+										{!isLastEntry && (
+											<span aria-hidden className={styles.divider}>
+												<Divider orientation="vertical" />
+											</span>
+										)}
+									</StackHorizontal>
+								</li>
+							);
+						}
+						const { label, as, ...entryProps } = entry;
+						return (
+							<li className={styles.entry} key={`${label}-${index}`}>
+								<StackHorizontal gap="S" align="center" wrap="nowrap">
+									<Link
+										{...entryProps}
+										as={as}
+										aria-current={isLastEntry ? 'page' : undefined}
+										withEllipsis
+									>
+										{label}
+									</Link>
+									{!isLastEntry && (
+										<span aria-hidden className={styles.divider}>
+											<Divider orientation="vertical" />
+										</span>
+									)}
+								</StackHorizontal>
+							</li>
+						);
+					})}
```

Comment:
this component is going to be really painful to maintain with all these duplicated code

Reply:
They are not duplicated, they each document precise usecases with strong types.